rest_framework/renderers.py

```diff
@@ -342,8 +342,12 @@ def render_field(self, field, parent_style):
         # Get a clone of the field with text-only value representation.
         field = field.as_form_field()
 
-        if style.get('input_type') == 'datetime-local' and isinstance(field.value, str):
-            field.value = field.value.rstrip('Z')
+        if style.get('input_type') == 'datetime-local':
+            # The format of an input type="datetime-local" is "yyyy-MM-ddThh:mm"
+            # followed by optional ":ss" or ":ss.SSS", so keep only the first three
+            # digits of milliseconds to avoid browser console error.
+            datetime_value = field._field.parent.validated_data.get(field.field_name)
+            field.value = datetime_value.replace(tzinfo=None).isoformat(timespec="milliseconds").rstrip('Z')
```

Comment:
It was harder than I thought!

Let me know if this is the correct way to get the datetime value.